packages/react-code-editor/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -226,485 +226,414 @@ export interface CodeEditorProps extends Omit<React.HTMLProps<HTMLDivElement>, '
-                overrideServices={overrideServices}
-                onChange={this.onChange}
-                onMount={this.editorDidMount}
-                theme={isDarkTheme ? 'vs-dark' : 'vs-light'}
-                loading={loading}
-                {...editorProps}
-              />
-            </div>
-          );
-
-          const hiddenFileInput = <input {...getInputProps()} /* hidden, necessary for react-dropzone */ hidden />;
-
-          return (
-            <div
-              className={css(
-                styles.codeEditor,
-                isReadOnly && styles.modifiers.readOnly,
-                isFullHeight && styles.modifiers.fullHeight,
-                className
+                  </EmptyStateActions>
+                  <EmptyStateActions>
+                    <Button variant="link" onClick={() => setShowEmptyState(false)}>
+                      {props.emptyStateLink}
+                    </Button>
+                  </EmptyStateActions>
+                </EmptyStateFooter>
               )}
-              ref={this.ref}
+            </EmptyState>
+          ) : (
+            <EmptyState
+              variant={EmptyStateVariant.sm}
+              titleText={props.emptyStateTitle}
+              icon={CodeIcon}
+              headingLevel="h4"
             >
-              {(isUploadEnabled || providedEmptyState) && !value ? (
-                <div
-                  {...getRootProps({
-                    onClick: (event) => event.stopPropagation() // Prevents clicking TextArea from opening file dialog
-                  })}
-                  className={css(styles.codeEditorContainer, isLoading && fileUploadStyles.modifiers.loading)}
-                >
-                  {editorHeader}
-                  <div className={css(styles.codeEditorMain, isDragActive && styles.modifiers.dragHover)}>
-                    {(showEmptyState || providedEmptyState) && !value ? (
-                      <div className={css(styles.codeEditorUpload)}>
-                        {hiddenFileInput}
-                        {emptyState}
-                      </div>
-                    ) : (
-                      <>
-                        {hiddenFileInput}
-                        {editor}
-                      </>
-                    )}
-                  </div>
-                </div>
-              ) : (
-                <>
-                  {editorHeader}
-                  {showEditor && <div className={css(styles.codeEditorMain)}>{editor}</div>}
-                </>
+              {!props.isReadOnly && (
+                <EmptyStateFooter>
+                  <EmptyStateActions>
+                    <Button variant="primary" onClick={() => setShowEmptyState(false)}>
+                      {props.emptyStateLink}
+                    </Button>
+                  </EmptyStateActions>
+                </EmptyStateFooter>
               )}
+            </EmptyState>
+          ));
+
+        const tooltipProps = {
```

Comment:
Let's move this out fo the JSX, just before the return statement. This doesn't really need to be here since it's not accessing anything scoped here.